### DISCUSSION TITLE
【vite+react】how can i build a basePath for the html when run npm stripts storybookbuild

### DISCUSSION THREAD

**Discussion by Henrycheheng** [storybookjs/storybook#29199] [Help]
### Summary

I use the vite+react project, what I want to ask is When I use storybook build this npm script to build, the storybook-static directory will be generated, 

and the js files referenced in the index.html are relative paths, like this: ./, 

I want to deploy this document application now, it will be published to npm pkg file has a path, 

**### So I want to add a common path to all references to js**, I tried ViteFinal, but I'm not sure if it works, can anyone help me?

**Comment by isaiahdaviscom** [comment]
# Solutions
You did not provide a way to reproduce the issue or a link to your code, so providing a solution based on your summarization was challenging. Here are some possible solutions to the problem.

Hope something helps.

Best regards,
Isaiah Davis

Github [isaiahdaviscom](https://github.com/isaiahdaviscom) | Website [isaiahdavis.com](https://isaiahdavis.com/)

## Direct:

``` javascript
module.exports = {
  stories: ['../src/**/*.stories.@(js|jsx|ts|tsx)'],
  addons: [],
  viteFinal: (config) => {
    // Set the base path for the built files
    config.base = '/your-base-path/';
    return config;
  },
};
```

## Alternative:

``` javascript
// package.json
"scripts": {
  "build-storybook": "build-storybook --static-dir ./storybook-static"
}
```

``` yml
# .env file
VITE_PUBLIC_PATH=/your-base-path/
```

``` javascript
export default {
  base: process.env.VITE_PUBLIC_PATH || '/',
};
// OR vite.config.js
export default {
  base: '/your-base-path/',
};
```

``` javascript
// .storybook/main.js
module.exports = {
  webpackFinal: async (config) => {
    config.output.publicPath = '/your-base-path/';
    return config;
  },
};
```
